### PR TITLE
fix: use detached context for CreateMember rollback

### PR DESCRIPTION
**Pull request by bramwelt**
## Summary

- `CreateMember` rollback deferred func was calling `deleteMemberKeys` with the HTTP request context, which is cancelled when the client times out (e.g. v1-sync-helper's 30-second `http.Client.Timeout` under onboarding load)
- NATS KV cleanup operations all fail immediately on a cancelled context, leaving the uniqueness key (`lookup/member/<sha256-hash>`) orphaned in the `committee-members` bucket
- Subsequent create attempts for the same email+committeeUID find the stale key via `UniqueMember` → `ErrKeyExists` → 409 Conflict, blocking the member from ever being created
- Fix: use `context.WithTimeout(context.Background(), 10s)` in the rollback, matching the pattern already used by `UpdateMember` for its stale-key cleanup goroutine

## Test plan

- [ ] All existing tests pass (`make test`)
- [ ] `make lint` clean
- [ ] `make build && make build-cli` clean
- [ ] Manual: verify a member blocked by a stale uniqueness key can be created after key removal

Closes LFXV2-2984

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/service/committee_member_writer.go [hunk 1]
```diff
@@ -101,7 +101,9 @@ func (uc *committeeWriterOrchestrator) CreateMember(ctx context.Context, member
 	)
 	defer func() {
 		if err := recover(); err != nil || rollbackRequired {
-			uc.deleteMemberKeys(ctx, keys, rollbackRequired)
+			cleanupCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+			uc.deleteMemberKeys(cleanupCtx, keys, rollbackRequired)
```
[the remark this answers is not in the document]
Comment (bramwelt) [reply internal/service/committee_member_writer.go:106]: Done. Added `TestCreateMember_RollbackSurvivesCancelledContext` in commit e4d8727.

The test:
- Adds a `rejectCancelledCtx` field to `TestMockCommitteeMemberWriter` so `DeleteMember`/`GetMemberRevision` return `ctx.Err()` when the context is done. Without this, the existing mock ignores context state and the test would pass with either the old or new code.
- Pre-seeds the uniqueness key in `memberWriter.members` so `GetMemberRevision` can resolve it during rollback (`UniqueMember` stores to `w.keys` but `GetMemberRevision` reads `w.members`).
- Cancels the caller context before calling `CreateMember` to simulate a timed-out HTTP client, forces `orgIndexErr` to trigger rollback at the org-index step, and asserts the uniqueness key appears in `deletedKeys`.

With the old code (passing the cancelled request context to `deleteMemberKeys`), `rejectCancelledCtx` would cause `GetMemberRevision` to return `ctx.Err()` immediately and the key would not be cleaned up. With the fix, the detached `cleanupCtx` is live and the assertion passes.

**Comment by bramwelt** [comment]
## Review Feedback Addressed

Commit: e4d8727 — test(member): regression for cancelled-ctx rollback

### Changes Made
- **internal/service/committee_member_writer_test.go**: Added `TestCreateMember_RollbackSurvivesCancelledContext` — a regression test that cancels the caller context before `CreateMember`, forces a later write failure, and asserts the uniqueness key is cleaned up using a live (detached) cleanup context (per copilot-pull-request-reviewer[bot])

### Threads Resolved
1 of 1 unresolved threads addressed in this iteration.

File: internal/service/committee_member_writer_test.go [hunk 2]
```diff
@@ -961,6 +968,69 @@ func TestCreateMember_OrgIndexWriteFailsRollsBack(t *testing.T) {
...
+	require.Error(t, err)
+
+	// Rollback must have cleaned up the uniqueness key using a live (detached) context.
+	// If rollback incorrectly used the cancelled request context, rejectCancelledCtx would
+	// cause GetMemberRevision/DeleteMember to return an error and the key would not appear
+	// in deletedKeys.
+	assert.True(t, memberWriter.wasDeleted(uniqueKey),
+		"rollback must remove the uniqueness key even when the request context is cancelled")
+}
+
```
[the remark this answers is not in the document]
Comment (bramwelt) [reply internal/service/committee_member_writer_test.go:1033]: Thanks for the review. Keeping this standalone for now.

The immediately adjacent `TestCreateMember_OrgIndexWriteFailsRollsBack` uses the same single-scenario pattern for the same reason: rollback tests require per-test infrastructure state (`rejectCancelledCtx`, pre-seeded `memberWriter.members[uniqueKey]`) that is unique to each failure path and does not generalize into table rows. Converting to a table with one entry would add struct boilerplate without making the test easier to extend — the next rollback-path regression would have its own distinct setup.

The table-driven guideline applies to tests that cover a range of input variations (valid inputs, edge cases, boundary values). This test has a fixed setup for a single regression scenario. Happy to convert if there is a second scenario that would benefit from the table structure.

**Comment by bramwelt** [comment]
## Review Feedback Addressed (iteration 2)

No new code changes — one thread responded to.

### No Change Needed
- **internal/service/committee_member_writer_test.go:971-1033**: The table-driven guideline applies to tests covering a range of input variations. This rollback regression test has exactly one meaningful scenario with per-test infrastructure state (`rejectCancelledCtx`, pre-seeded uniqueness key) that is specific to this failure path. The immediately adjacent `TestCreateMember_OrgIndexWriteFailsRollsBack` uses the same standalone pattern. Converting to a table with one row adds boilerplate without benefit (flagged by coderabbitai[bot])

### Threads Resolved
0 of 1 thread resolved in this iteration — leaving open for reviewer confirmation.

**Review by andrest50 — APPROVED** [review]
